### PR TITLE
fix: fix some SAML app bugs

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

fix some SAML app bugs
1. fix the way of building IdP `entityId` (resolves LOG-10560)
2. create an `active` secret on SAML app creation (previously we create a inactive secret and manually activation is required) (resolves LOG-10558)
3. expand the `config` (includes `attributeMapping`, `acsUrl` and `entityId`) in POST and PATCH APIs (resolves LOG-10556)
4. update `SamlAcsUrl` type, `binding` field should be required. (resolves LOG-10561)
5. add `fingerprints` to SAML app secrets APIs response (resolves LOG-10563)

## Testing

Update integration tests.

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [x] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/saml-applications/libraries/saml-applications.ts [hunk 1]
```diff
@@ -33,11 +34,15 @@ export const createSamlApplicationsLibrary = (queries: Queries) => {
...
-    lifeSpanInDays = 365 * 3
-  ): Promise<SamlApplicationSecret> => {
+  const createSamlApplicationSecret = async ({
+    applicationId,
+    lifeSpanInDays = 365 * 3,
+    active = false,
+  }: {
+    applicationId: string;
+    lifeSpanInDays?: number;
+    active?: boolean;
```
Comment (simeng-li) [inline packages/core/src/saml-applications/libraries/saml-applications.ts:44]:
```suggestion
    isActive?: boolean;
```

File: packages/core/src/saml-applications/libraries/saml-applications.ts [hunk 2]
```diff
@@ -86,7 +92,7 @@ export const createSamlApplicationsLibrary = (queries: Queries) => {
       Object.keys(applicationData).length > 0
         ? updateApplicationById(id, removeUndefinedKeys(applicationData))
         : originalApplication,
-      config
+      Object.keys(config).length > 0
```
Comment (simeng-li) [inline packages/core/src/saml-applications/libraries/saml-applications.ts:95]: Will this partially update the config values?  E.g. { entityId: XXX }

File: packages/core/src/saml-applications/libraries/saml-applications.ts [hunk 3]
```diff
@@ -33,11 +34,15 @@ export const createSamlApplicationsLibrary = (queries: Queries) => {
     },
   } = queries;
 
-  const createSamlApplicationSecret = async (
-    applicationId: string,
-    // Set certificate life span to 3 years by default.
-    lifeSpanInDays = 365 * 3
-  ): Promise<SamlApplicationSecret> => {
+  const createSamlApplicationSecret = async ({
```
Comment (simeng-li) [inline packages/core/src/saml-applications/libraries/saml-applications.ts:37]: add some comments for the input parameters. Especially for the `active`, there is a need to notify all the current active secrets that will be deactivated.

File: packages/core/src/saml-applications/routes/index.ts [hunk 4]
```diff
@@ -255,7 +262,10 @@ export default function samlApplicationRoutes<T extends ManagementApiRouter>(
         await updateSamlApplicationSecretStatusByApplicationIdAndSecretId(id, secretId, active);
 
       ctx.status = 200;
-      ctx.body = updatedSamlApplicationSecret;
+      ctx.body = {
+        ...updatedSamlApplicationSecret,
+        fingerprints: calculateCertificateFingerprints(updatedSamlApplicationSecret.certificate),
```
Comment (simeng-li) [inline packages/core/src/saml-applications/routes/index.ts:267]: Should it be plural?

File: packages/core/src/saml-applications/routes/index.ts [hunk 5]
```diff
@@ -266,9 +276,9 @@ export default function samlApplicationRoutes<T extends ManagementApiRouter>(
     koaGuard({
       params: z.object({ id: z.string() }),
       status: [200, 400, 404],
-      response: z.object({
-        certificate: z.string(),
-        fingerprints: certificateFingerprintsGuard,
+      response: samlApplicationSecretResponseGuard.pick({
```
Comment (simeng-li) [inline packages/core/src/saml-applications/routes/index.ts:279]: What is the use case of this API, do we need to return the fingerprints here? Also should be `fingerprint`?

File: packages/core/src/saml-applications/libraries/saml-applications.ts (hunk 2 above)
Comment (darcyYe) [reply packages/core/src/saml-applications/libraries/saml-applications.ts:103]: Add integration test to cover this.

File: packages/core/src/saml-applications/routes/index.ts (hunk 5 above)
Comment (darcyYe) [reply packages/core/src/saml-applications/routes/index.ts:279]: Discussed offline, updated

**Review by simeng-li — APPROVED** [review]

**Pull request merged**